statsmodels/tsa/statespace/kalman_filter.py

```diff
@@ -1253,36 +1258,35 @@ def update_filter(self, kalman_filter):
                 # produce forecasts, but forecast errors and the forecast
                 # error covariance matrix will be zeros - make them nan to
                 # improve clarity of results.
-                if self.nmissing[t] == self.k_endog:
+                if self.nmissing[t] > 0:
+                    mask = ~self.missing[:, t].astype(bool)
```

Comment:
If it's missing is integers, then it will be interpreted as index to rows or columns.
You need `bool` if you want boolean indexing with a mask. 
(Or use missing as index for the missing values, but that won't work for ndim > 1, if there are not equal numbers of missing.)

Is missing a 0, 1 array to work more easily in cython or could missing directly be stored as boolean array? (My first guess was that missing is already boolean.